java-squid/src/test/java/org/sonar/java/model/JavaTreeModelTest.java

```diff
@@ -1950,9 +2033,14 @@ public void array_access_expression() {
     String code = "class T { T() { return a[42]; } }";
     ArrayAccessExpressionTree tree = (ArrayAccessExpressionTree) expressionOfReturnStatement(code);
     assertThat(tree.is(Tree.Kind.ARRAY_ACCESS_EXPRESSION)).isTrue();
+    assertThatChildrenIteratorHasSize(tree, 2);
     assertThat(tree.expression()).isNotNull();
-    assertThat(tree.index()).isNotNull();
-    assertThatChildrenIteratorHasSize(tree, 4);
+    ArrayDimensionTree dimension = tree.dimension();
+    assertThat(dimension).isNotNull();
+    assertThat(dimension.openBracketToken().text()).isEqualTo("[");
+    assertThat(dimension.expression().is(Tree.Kind.INT_LITERAL));
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Complete the assertion. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS2970)